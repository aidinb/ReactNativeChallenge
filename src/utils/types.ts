export interface Address {
    city: string;
    street: string;
    suite: string;
    zipcode: string;
    geo: object[]; // Assuming geo is an array of objects, adjust if necessary
}

export interface Company {
    bs: string;
    catchPhrase: string;
    name: string;
}

export interface Item {
    id: number;
    name: string;
    username: string;
    email: string;
    phone: string;
    website: string;
    address: Address;
    company: Company;
}
