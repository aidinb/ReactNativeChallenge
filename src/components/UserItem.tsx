import React from 'react';
import { TouchableOpacity, Text, StyleSheet, Dimensions } from 'react-native';
import { Item } from '../utils/types.ts';

interface UserItemProps {
  item: Item;
  onPress: (item: Item) => void;
}

const { width } = Dimensions.get('window');

const UserItem: React.FC<UserItemProps> = ({ item, onPress }) => (
    <TouchableOpacity
        onPress={() => onPress(item)}
        style={styles.card}
    >
      <Text style={styles.text}>{item.name}</Text>
    </TouchableOpacity>
);

const styles = StyleSheet.create({
  card: {
    width: width * 0.9,
    paddingVertical: 15,
    backgroundColor: '#ffffff',
    marginVertical: 10,
    borderRadius: 8,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.2,
    shadowRadius: 4,
    elevation: 2,
    alignSelf: 'center',
  },
  text: {
    fontSize: 16,
    color: '#333',
    paddingLeft: 15,
  },
});

export default React.memo(UserItem);
