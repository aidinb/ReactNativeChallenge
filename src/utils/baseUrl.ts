import axios, {AxiosInstance} from 'axios';

const baseUrlConfig: AxiosInstance = axios.create({
  baseURL: 'https://jsonplaceholder.typicode.com',
  timeout: 20000,
  headers: {
    'Content-Type': 'application/json',
  },
});

export const baseAPI = baseUrlConfig;
