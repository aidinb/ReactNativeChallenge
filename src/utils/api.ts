import {baseAPI} from './baseUrl.ts';

export const getItems = (): Promise<any> =>
  baseAPI.get('/users').then(res => res.data);
