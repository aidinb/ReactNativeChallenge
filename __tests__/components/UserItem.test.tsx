import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import UserItem from '../../src/components/UserItem'; // Ensure the correct import path
import { Item } from '../../src/utils/types'; // Import the Item type

describe('UserItem Component', () => {
    const mockOnPress = jest.fn(); // Mock function to track if it's called

    const user: Item = {
        id: 1,
        name: 'Leanne Graham',
        username: 'Bret',
        email: 'Sincere@april.biz',
        phone: '1-770-736-8031 x56442',
        website: 'hildegard.org',
        address: {
            city: 'Gwenborough',
            geo: [{ lat: '1', lng: '1' }], // Assuming geo is an array of objects
            street: 'Kulas Light',
            suite: 'Apt. 556',
            zipcode: '92998-3874',
        },
        company: {
            bs: 'harness real-time e-markets',
            catchPhrase: 'Multi-layered client-server neural-net',
            name: 'Romaguera-Crona',
        },
    };

    it('renders correctly and displays user name', () => {
        const { toJSON, getByText } = render(<UserItem item={user} onPress={mockOnPress} />);

        expect(toJSON()).toMatchSnapshot();

        expect(getByText('Leanne Graham')).toBeTruthy();
    });

    it('calls onPress with the correct user when pressed', () => {
        const { getByText } = render(<UserItem item={user} onPress={mockOnPress} />);

        // Simulate pressing the TouchableOpacity
        fireEvent.press(getByText('Leanne Graham'));

        // Verify that the onPress callback was called with the correct item
        expect(mockOnPress).toHaveBeenCalledWith(user);
    });
});
