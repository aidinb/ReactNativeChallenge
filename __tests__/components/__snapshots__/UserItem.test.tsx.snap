// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserItem Component renders correctly and displays user name 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "elevation": 2,
      "marginVertical": 10,
      "opacity": 1,
      "paddingVertical": 15,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
      "width": 675,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "paddingLeft": 15,
      }
    }
  >
    Leanne Graham
  </Text>
</View>
`;
